atom/browser/net/atom_navigation_throttle.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/net/atom_navigation_throttle.h"
+
+#include "atom/browser/api/atom_api_web_contents.h"
+#include "content/public/browser/navigation_handle.h"
+
+namespace atom {
+
+AtomNavigationThrottle::AtomNavigationThrottle(
+    content::NavigationHandle* navigation_handle)
+    : content::NavigationThrottle(navigation_handle) {}
+
+AtomNavigationThrottle::~AtomNavigationThrottle() {}
+
+content::NavigationThrottle::ThrottleCheckResult
+AtomNavigationThrottle::WillRedirectRequest() {
+  auto* handle = navigation_handle();
+  auto* contents = handle->GetWebContents();
+  if (!contents)
+    return PROCEED;
```

Comment:
Yeah this is more like a `NOREACHED()` but I thought I'd handle it for safety.  Current behavior where we have no throttles would be for the navigation to proceed so I'd like to maintain that behavior.  Might be worth adding a `DCHECK` or something there so that it would cause failures in debug mode though?

Reply:
Its better to turn this into a `DCHECK`, if such a navigation occurs might be of interest to the chromium team. But if `api_web_contents` is not available then it should be terminated, in the past we have been caught by crashes where users destroy webcontents inside a navigation event. Although the destruction is made async, I would prefer terminating the navigation in such scenarios to be extra secure. Chromium doesn't worry about those cases as there is no direct user script intervention in the browser process.

Reply:
> Current behavior where we have no throttles would be for the navigation to proceed so I'd like to maintain that behavior

👍